sympy/solvers/solveset.py

```diff
@@ -3239,6 +3241,15 @@ def _solve_using_known_values(result, solver):
                         soln = solver(eq2, sym)
                         total_solvest_call += 1
                         soln_new = S.EmptySet
+                        if isinstance(soln, ConditionSet):
+                            if soln.args[2] in (S.Reals, S.Complexes):
+                                soln = S.EmptySet
+                                # don't do `continue` we may get soln
+                                # in terms of other symbol(s)
+                                not_solvable = True
+                                total_conditionst += 1
+                            else:
+                                soln = soln.args[2]
```

Comment:
Whereas for the cases where the `conditionset` is not returning the result will give domain at `ConditionSet(...).args[2]`. For which this code is to handle such situations.

Since `nonlinsolve` works with only `S.Reals` and `S.Complexes` so I've considered these 2 cases only.

https://github.com/sympy/sympy/blob/f4c2dce477141ac1b79fea79ed71692988bbb4bf/sympy/solvers/solveset.py#L3327-L3333